src/web/services/search.service.ts

```diff
@@ -169,47 +157,44 @@ export class SearchService {
   }
 
   getPrivileges(
-    coursesWithSections: SearchStudentsTable[],
+    coursesWithStudents: SearchStudentsListRowTable[],
   ): Observable<InstructorPrivilege[]> {
-    if (coursesWithSections.length === 0) {
+    if (coursesWithStudents.length === 0) {
       return of([]);
     }
-    return forkJoin(
-      coursesWithSections.map((course: SearchStudentsTable) => {
-        return course.sections.map((section: StudentListSectionData) => {
-          return this.instructorService.loadInstructorPrivilege({
-            courseId: course.courseId,
-            sectionName: section.sectionName,
-          });
-        });
-      }).reduce(
-        (acc: Observable<InstructorPrivilege>[], val: Observable<InstructorPrivilege>[]) =>
-        acc.concat(val),
-        [],
-      ),
-    );
+    const privileges: any[] = [];
```

Comment:
Try not to use `any[]` as the type. You should explicitly state the variable type where possible

Reply:
Hi, I have reflected your feedback on my new commit. I also have another version that uses chaining method (forEach) rather than the for-loops. I think using forEach is easier to read, but I would like to ask for your opinion. 
In case you want to see the new version, here is a screenshot of it. 
If you think this is better, I will push it. Thank you!
![image](https://user-images.githubusercontent.com/32270970/87252142-6e896e80-c436-11ea-9722-a582db2f8b88.png)

Reply:
Yes, this version is easier to read. You can go ahead with this version instead